ocp_utilities/monitoring.py

```diff
@@ -0,0 +1,227 @@
+        response = requests.get(
+            f"{self.api_url}{query}", headers=self.headers, verify=self.verify_ssl
+        )
+
+        try:
+            return json.loads(response.content)
+        except JSONDecodeError as json_exception:
+            LOGGER.error(
+                "Exception converting query response to JSON: "
+                f"exc={json_exception} response_status_code={response.status_code} response={response.content}"
+            )
+            raise
+
+    def query(self, query):
+        """
+        get the prometheus query result
+
+        Args:
+            query (str): promthetheus query string
+
+        Returns:
+            dict: query result
+        """
+        return self._get_response(query=f"{self.api_v1}/query?query={query}")
+
+    def get_all_alerts_by_alert_name(self, alert_name):
+        """
+        Get alert by alert name if it's an active alert
+
+        Args:
+             alert (str): alert name
+
+        Examples:
+             result = prometheus.get_alert(alert='WatchDog')
+
+        Returns:
+             list: list containing alert metrics
+        """
+        response = self.alerts()
+        alert_list = []
+        for alert in response["data"]["alerts"]:
+            if alert and alert["labels"]["alertname"] == alert_name:
+                alert_list.append(alert)
+        return alert_list
+
+    def get_firing_alerts(self, alert_name):
+        """
+        get all the firing alerts from list of active alerts
+        """
+        alert_list = self.get_all_alerts_by_alert_name(alert_name=alert_name)
+        firing_alerts = []
+        for alert in alert_list:
+            if alert["state"] == "firing":
+                firing_alerts.append(alert)
+
+        return firing_alerts
+
+    def wait_for_firing_alert_sampler(self, alert_name, timeout=TIMEOUT_10MIN):
+        """
+        Sample output for an alert if found in fired state
+
+        Args:
+             alert (str): alert name
+             timeout (int): wait time, default is 10 mins
+
+        Return:
+             sample (list): list of all alerts that match the alert name and in firing state
+
+        Raise:
+             TimeoutExpiredError: if alert is not fired before wait_timeout
+        """
+        sampler = TimeoutSampler(
+            wait_timeout=timeout,
+            sleep=self.scrape_interval,
+            func=self.get_firing_alerts,
+            alert_name=alert_name,
+        )
+        sample = None
+        try:
+            for sample in sampler:
+                if sample:
+                    LOGGER.info(f"Found alert: {alert_name} in firing state.")
+                    return sample
+
+        except TimeoutExpiredError:
+            LOGGER.error(f"{alert_name} currently not in firing state")
+            raise
+
+    def get_scrape_interval(self):
+        """
+        get prometheus scrap interval
+
+        Returns:
+             int: scrape time interval
```

Comment:
Please add if not found return default 30

Reply:
done